### ISSUE TITLE
Auto-completion ruins my import line after typing "flash."

### ISSUE THREAD

**Issue by starry-abyss** [fdorg/flashdevelop#949]
In FlashDevelop 5.0.2.2
![flashdevelop-autocomplete](https://cloud.githubusercontent.com/assets/11571820/11021863/4a203d1c-8660-11e5-85bf-656ee080828e.gif)

When I'm about to type in the editor "import flash.system.System;" it's altered to "import haxe.remoting.FlashJsConnection." after typing "flash.".

It's shown in the attachment gif. I've tried different haxe auto-completion settings, but it either stays the same or auto-completion turns off entirely.

Any clue?

**Comment by elsassph** [comment]
I see, a workaround is to press Escape to dismiss the completion list.

If you're using Flixel with OpenFL you may have the flash package back by choosing the Flash target in the main toolbar.

**Comment by starry-abyss** [comment]
Thanks, Escape is better than nothing!

Not sure, what you mean by "have the flash package back", but I noticed now if I type "flash", then hit Escape, then period, then Ctrl+Space, I can get the class list from Flash package! But still even after choosing Flash target in toolbar - if not pressing Escape - I get the behaviour as in picture above...

**Comment by elsassph** [comment]
OK that's a bit strange, normally OpenFl should provide definitions for the Flash API. If you type 'new Sprite' do you see 'flash.display.Sprite' in the completion list?

**Comment by starry-abyss** [comment]
I get 'openfl.display.Sprite' then. Maybe it has something to do with HaxeFlixel doing "--remap flash:openfl" under the hood?

**Comment by elsassph** [comment]
Right, the thing is that by default OpenFL gives you the Flash API as 'openfl.*'.

It's just typedefs (aliases) when you target Flash, and it has a custom implementation for other platforms: that's the preferred way to use the Flash API, unless the classes are really Flash specific.

**Comment by starry-abyss** [comment]
Got it. Thanks!

**Comment by wise0704** [comment]
Probs close this issue?

**Comment by starry-abyss** [comment]
I think it's still a bug, though I may not personally need it fixed (but may test the fix if it comes out)

**Comment by Neverbirth** [comment]
I don't think this is a bug either. The thing is that you are trying to enter a non-existent element (which doesn't mean there is some other sort of bug), and the behaviour is the one you see, but it's that way on purpose. Other IDEs do other things like completely hide the list, or leave some possible results but without selecting them... I sincerely don't like any of them (but better the second than the first), another option perhaps would be to use a hybrid approach: if the difference with the existing possible results is not very big leave the current behaviour, if not use some of the other ways... maybe it could be a setting.

**Comment by starry-abyss** [comment]
If there was a setting for making auto-completion wait for user to agree with the proposed variant it'd be great (e.g. one would have to press Enter). Or is there one?

**Comment by Neverbirth** [comment]
That wouldn't be very nice heh. At the end of the day you save thousands of keystrokes by making the dot insert the selected ítem and instantly display the members of an object.

**Comment by wise0704** [comment]
For `import` statements, I don't think there will be any case when you want to write something that is not in the auto completion. If there is, well then it's a different problem because it means FD is not recognising some packages. 

Still, if by any chance, there is a case when you need to write something not in the auto completion, and it's not a bug in FD either (very weird case, i think), pressing ESC sounds much better than modifying the program just for those very rare cases.
